core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -202,15 +202,24 @@ private boolean waitForChannelReady(ManagedChannel managedChannel, long healthCh
   /**
    * Shuts down the managed channel.
    */
-  private void shutdownManagedChannel(ManagedChannel managedChannel, long shutdownTimeoutMs) {
-    managedChannel.shutdown();
+  private void shutdownManagedChannel(ManagedChannel managedChannel, boolean shutdownNow,
+      long shutdownTimeoutMs) {
+    // Shutdown channel forcefully if requested.
+    if (shutdownNow) {
+      managedChannel.shutdownNow();
```

Comment:
Do we need to run the following code if `shutdownNow` is true?

Reply:
I guess we'd still want to guarantee complete termination before allowing another client to potentially open a channel to same target.

From `ManagedChannel#shutdownNow()` java doc:

   * Initiates a forceful shutdown in which preexisting and new calls are cancelled. Although
   * forceful, the shutdown process is still not instantaneous; {@link #isTerminated()} will likely
   * return {@code false} immediately after this method returns.

Reply:
Makes sense, is the idea that `isTerminated` will return true when this method returns? We handle the two calls to `shutdownNow` in this method differently (we do not await termination again in finally)

Reply:
Yes. Updated the method to avoid multiple `shutdownNow()`s.